metrics-operator/api/v1alpha3/keptnmetric_webhook_test.go

```diff
@@ -0,0 +1,49 @@
+package v1alpha3
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+func TestKeptnMetric_validateRangeInterval(t *testing.T) {
+
+	tests := []struct {
+		name string
+		Spec KeptnMetricSpec
+		want *field.Error
+	}{
+		{
+			name: "bad-provider",
+			Spec: KeptnMetricSpec{
+				Range: RangeSpec{Interval: "5mins"},
+			},
+			want: field.Invalid(
+				field.NewPath("spec").Child("range").Child("interval"),
+				"5mins",
+				errors.New("Forbidden! The time interval cannot be parsed. Please check for suitable conventions").Error(),
+			),
+		},
+
```

Comment:
I would also add a case for empty string

Reply:
Just wanted to confirm some things, I think some things are going wrong, please share your views on this @thisthat @bacherfl 
- During the below scenario, we are getting an error. 
```
range:
  interval: ""
```
- During this scenario, the resource gets created, I think the deafult value is being taken over here i.e `5m`
```
range:
   interval: 
```
- However if we dont provide the `range.interval` field, we are still getting error, I think this is wrong right? as we are supposed to make this `range.interval` feature optional.

Reply:
Hi @rakshitgondwal - indeed, the range spec should be optional, so in my opinion we should not get an error if we do not provide it. Can you try to make the `Range` property of the `KeptnMetric` struct a pointer? I think this way we can keep this new property optional, and in the webhook we do the validation only if this property is `!= nil`